geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRemRangeByLexIntegrationTest.java

```diff
@@ -0,0 +1,329 @@
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_MIN_MAX_NOT_A_VALID_STRING;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.BIND_ADDRESS;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.REDIS_CLIENT_TIMEOUT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.Protocol;
+
+import org.apache.geode.redis.RedisIntegrationTest;
+
+@RunWith(JUnitParamsRunner.class)
+public abstract class AbstractZRemRangeByLexIntegrationTest implements RedisIntegrationTest {
+  public static final String KEY = "key";
+  public static final int SCORE = 1;
+  public static final String BASE_MEMBER_NAME = "v";
+
+  JedisCluster jedis;
+
+  @Before
+  public void setUp() {
+    jedis = new JedisCluster(new HostAndPort(BIND_ADDRESS, getPort()), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void shouldError_givenWrongNumberOfArguments() {
+    assertExactNumberOfArgs(jedis, Protocol.Command.ZREMRANGEBYLEX, 3);
+  }
+
+  @Test
+  @Parameters({"a", "--", "++", "4"})
+  public void shouldError_givenInvalidMinOrMax(String invalidArgument) {
+    assertThatThrownBy(() -> jedis.zremrangeByLex("fakeKey", invalidArgument, "+"))
+        .hasMessageContaining(ERROR_MIN_MAX_NOT_A_VALID_STRING);
+    assertThatThrownBy(() -> jedis.zremrangeByLex("fakeKey", "-", invalidArgument))
+        .hasMessageContaining(ERROR_MIN_MAX_NOT_A_VALID_STRING);
+    assertThatThrownBy(() -> jedis.zremrangeByLex("fakeKey", invalidArgument, invalidArgument))
+        .hasMessageContaining(ERROR_MIN_MAX_NOT_A_VALID_STRING);
+  }
+
+  @Test
+  public void shouldReturnZero_givenNonExistentKey() {
+    jedis.zadd(KEY, SCORE, "member1");
+    assertThat(jedis.zremrangeByLex("fakeKey", "-", "+")).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldReturnZero_givenMinGreaterThanMax() {
+    jedis.zadd(KEY, SCORE, "member");
+
+    // Range + <= member name <= -
+    assertThat(jedis.zrangeByLex(KEY, "+", "-")).isEmpty();
+    // Range z <= member name <= a
+    assertThat(jedis.zrangeByLex(KEY, "[z", "[a")).isEmpty();
+  }
+
+  @Test
+  public void shouldReturnOne_givenOneMemberAndMemberNameInRangeInclusiveMtoN() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range m <= member name <= n
+    assertThat(jedis.zremrangeByLex(KEY, "[m", "[n")).isEqualTo(1);
```

Comment:
Very minor thing, but `isEqualTo(1)` can be simplified to `isOne()` throughout this class.